ledgerwallet/manifest_json.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+from typing import Dict, Optional
+
+from ledgerwallet import params
+from ledgerwallet.manifest import AppManifest, icon_from_file
+
+
+class AppManifestJson(AppManifest):
+    def __init__(self, filename):
+        with open(filename) as f:
+            self.path = os.path.dirname(filename)
+            self.dic = json.load(f)
+            assert "targetId" in self.dic and "binary" in self.dic
+
+    @property
+    def app_name(self) -> str:
+        return self.dic["name"]
+
+    def data_size(self, device) -> int:
+        if "dataSize" not in self.dic:
+            return 0
+        else:
+            return self.dic["dataSize"]
+
+    def get_application_flags(self, device) -> int:
+        if "flags" not in self.dic:
+            return 0
+        else:
+            return int(self.dic["flags"], 16)
+
+    def get_api_level(self, device) -> int:
```

Comment:
Instead of having two methods `has_api_level` and `get_api_level`, you could have one method with an optional output:
```python
    def get_api_level(self, device) -> Optional[int]:
        if "apiLevel" not in self.dic:
            return None
        return int(self.dic["apiLevel"], 10)
````